uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/LeaveRequestTest.php

```diff
@@ -3749,6 +3749,162 @@ public function testDeleteAttachmentShouldThrowAnExceptionIfAttachmentIDIsMissin
+    // Contact2
+    $result = civicrm_api3('LeaveRequest', 'getBreakdown', [
+      'leave_request_id' => $leaveRequest->id,
+      'check_permissions' => true,
+    ]);
+    $this->assertEmpty($result['values']);
+
+    $this->registerCurrentLoggedInContactInSession($contact2['id']);
+
+    // Contact2 should be able to get the breakdown for their own leave request
+    $result = civicrm_api3('LeaveRequest', 'getBreakdown', [
+      'leave_request_id' => $leaveRequest->id,
+      'check_permissions' => true,
+    ]);
+    $this->assertCount(3, $result['values']);
+  }
+
+  public function testGetBreakdownShouldReturnEmptyIfAManagerTriesToGetTheBreakdownOfSomeoneWhoTheyDontManage() {
+    $manager = ContactFabricator::fabricate();
+    $contact1 = ContactFabricator::fabricate();
+    $contact2 = ContactFabricator::fabricate();
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $contact2['id']],
+      ['period_start_date' => CRM_Utils_Date::processDate('+5 days')]
+    );
+
+    $leaveRequest = LeaveRequestFabricator::fabricateWithoutValidation([
+      'type_id' => 1,
+      'contact_id' => $contact2['id'],
+      'from_date' => CRM_Utils_Date::processDate('+5 days'),
+      'to_date' =>  CRM_Utils_Date::processDate('+7 days'),
+    ], true);
+
+    $this->registerCurrentLoggedInContactInSession($manager['id']);
+    $this->setContactAsLeaveApproverOf($manager, $contact1);
+    $this->setPermissions(['access AJAX API']);
+
+    // Manager only manages Contact 1, so they should not be able to get the
+    // breakdown for a leave request of Contact2
+    $result = civicrm_api3('LeaveRequest', 'getBreakdown', [
+      'leave_request_id' => $leaveRequest->id,
+      'check_permissions' => true,
+    ]);
+    $this->assertEmpty($result['values']);
+  }
+
+  public function testGetBreakdownShouldNotReturnEmptyIfAManagerTriesToGetTheBreakdownOfSomeoneTheyManage() {
+    $manager = ContactFabricator::fabricate();
+    $contact1 = ContactFabricator::fabricate();
+    $contact2 = ContactFabricator::fabricate();
+
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $contact2['id']],
+      ['period_start_date' => CRM_Utils_Date::processDate('+5 days')]
+    );
+
+    $leaveRequest = LeaveRequestFabricator::fabricateWithoutValidation([
+      'type_id' => 1,
+      'contact_id' => $contact2['id'],
+      'from_date' => CRM_Utils_Date::processDate('+5 days'),
+      'to_date' =>  CRM_Utils_Date::processDate('+7 days'),
+    ], true);
+
+    $this->registerCurrentLoggedInContactInSession($manager['id']);
+    $this->setContactAsLeaveApproverOf($manager, $contact2);
+    $this->setPermissions(['access AJAX API']);
+
+    $result = civicrm_api3('LeaveRequest', 'getBreakdown', [
+      'leave_request_id' => $leaveRequest->id,
+      'check_permissions' => true,
+    ]);
+    $this->assertCount(3, $result['values']);
+  }
+
+  public function testGetBreakdownShouldNotReturnIfAnAdminTriesToAccessTheBreakdownOfAnyLeaveRequest() {
```

Comment:
This should probably be `testGetBreakdownShouldNotReturnEmptyIfAnAdminTriesToAccessTheBreakdownOfAnyLeaveRequest` or `testGetBreakdownReturnsResultsIfAnAdminTriesToAccessTheBreakdownOfAnyLeaveRequest`